src/components/pages/Home.tsx

```diff
@@ -46,26 +44,43 @@ function SectionQueryResult({
 }) {
   const { config } = useOcean()
   const [result, setResult] = useState<QueryResult>()
+  const [dids, setDIDs] = useState<string>()
 
   useEffect(() => {
     if (!config?.metadataCacheUri) return
-
     const source = axios.CancelToken.source()
+    getHighestLiquidityDIDs().then((results) => {
+      setDIDs(results)
+    })
 
     async function init() {
-      const result = await queryMetadata(
-        query,
-        config.metadataCacheUri,
-        source.token
-      )
-      setResult(result)
+      try {
+        const result = await queryMetadata(
+          query,
+          config.metadataCacheUri,
+          source.token
+        )
+        if (result.totalResults <= 15) {
+          const searchDIDs = dids.split(' ')
+          const sortedAssets = sortElements(result.results, searchDIDs)
+          // We take more assets than we need from the subgraph (to make sure
+          // all the 9 assets with highest liquidity we need are in OceanDB)
+          // so we need to get rid of the surplus
+          const overflow = sortedAssets.length - 9
+          sortedAssets.splice(sortedAssets.length - overflow, overflow)
+          result.results = sortedAssets
+        }
+        setResult(result)
+      } catch (error) {
+        Logger.log(error.message)
+      }
     }
     init()
 
     return () => {
       source.cancel()
     }
-  }, [config?.metadataCacheUri, query])
+  }, [config?.metadataCacheUri, query, dids])
```

Comment:
dependencies here look ok so I would assume `query` does not get updated upon network switch. As a test could try moving the whole query into state instead of just the DIDs https://github.com/oceanprotocol/market/pull/512/files#diff-1df2f2876c20c90c294da5abb9fcf0812eec3448de2aea5eb20e5e5f4a7d6cc7R104-R113

This might solve some update issues upon network change